src/Web/Controllers/AccountController.cs

```diff
@@ -73,11 +86,38 @@ public async Task<IActionResult> SignIn(LoginViewModel model, string returnUrl =
             }
             ViewData["ReturnUrl"] = returnUrl;
 
-            var result = await _signInManager.PasswordSignInAsync(model.Email, model.Password, model.RememberMe, lockoutOnFailure: false);
-            if (result.Succeeded)
+            ApplicationUser applicationUser = new ApplicationUser()
+            {
+                UserName = model.Email
+            };
+            var passwordHash = _passwordHasher.HashPassword(applicationUser, model.Password);
+
+            var deviceContext = new DeviceContext
+            {
+                IPAddress = _contextAccessor.HttpContext.Connection.RemoteIpAddress.ToString()
+            };
+
+            SignInRequest req = new SignInRequest()
             {
-                await TransferBasketToEmailAsync(model.Email);
-                return RedirectToLocal(returnUrl);
+                SignInId = Guid.NewGuid().ToString(),
+                PasswordHash = passwordHash,
+                MerchantLocalDate = DateTimeOffset.Now,
+                CustomerLocalDate = model.DeviceFingerPrinting.ClientDate,
+                UserId = model.Email,
+                DeviceContextId = model.DeviceFingerPrinting.SessionId,
+                AssessmentType = AssessmentType.Protect.ToString(),
+                CurrentIpAddress = deviceContext.IPAddress
+            };
+
+            var signInAssessmentResponse = await _fraudProtectionService.PostSignIn(req, _fraudProtectionService.NewCorrelationId);
+            if (signInAssessmentResponse != null && signInAssessmentResponse.ResultDetails["MerchantRuleDecision"].Equals("Approve"))
```

Comment:
Can you create a response class similar to the purchase response and signup response classes?  You shouldn't need to do dictionary access like this.